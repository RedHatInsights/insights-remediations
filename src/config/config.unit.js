'use strict';

const path = require('path');

describe('Configuration', () => {
    test('Verify app-common-js function', () => {
        process.env.ACG_CONFIG = path.resolve('src/config/test.json');
        const config = require('.');

        expect(config).toMatchSnapshot();
    });
});
