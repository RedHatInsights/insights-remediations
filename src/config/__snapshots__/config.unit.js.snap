// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Configuration Verify app-common-js function 1`] = `
{
  "advisor": {
    "auth": "",
    "host": "http://insights-advisor-api.advisor-ci.svc.cluster.local:8000",
    "impl": undefined,
    "insecure": false,
    "revalidationInterval": 3600,
  },
  "bodyParserLimit": "1mb",
  "bop": {
    "host": "http://undefined:undefined",
    "impl": undefined,
    "insecure": false,
  },
  "cache": {
    "revalidationInterval": 600,
    "ttl": 86400,
  },
  "commit": undefined,
  "compliance": {
    "host": "http://compliance-backend.compliance-ci.svc.cluster.local:3000",
    "impl": undefined,
    "insecure": false,
    "revalidationInterval": 3600,
  },
  "configManager": {
    "auth": undefined,
    "host": "http://config-manager-service.config-manager-ci.svc.cluster.local:8081",
    "impl": undefined,
    "insecure": false,
    "revalidationInterval": 3600,
  },
  "contentServer": {
    "auth": "",
    "host": "http://insights-advisor-api.advisor-ci.svc.cluster.local:8000",
    "impl": undefined,
    "insecure": true,
    "revalidationInterval": 3600,
  },
  "db": {
    "benchmark": true,
    "database": "remediationstest",
    "define": {
      "charset": "utf8",
      "timestamps": false,
      "underscored": true,
    },
    "dialect": "postgres",
    "dialectOptions": {
      "ssl": {
        "ca": "=== CERTIFICATE ===",
      },
    },
    "host": "hostname",
    "logging": false,
    "password": "adminpassword",
    "pool": {
      "max": 50,
      "min": 5,
    },
    "ssl": true,
    "username": "adminusername",
  },
  "demo": false,
  "dispatcher": {
    "auth": "",
    "host": "http://playbook-dispatcher-api.playbook-dispatcher-ci.svc.cluster.local:8000",
    "impl": undefined,
    "insecure": false,
    "pageSize": 50,
    "revalidationInterval": 3600,
  },
  "env": "test",
  "fifi": {
    "text_update_full": true,
    "text_update_interval": 5000,
    "text_updates": true,
  },
  "inventory": {
    "host": "http://insights-inventory.platform-ci.svc.cluster.local:8080",
    "impl": undefined,
    "insecure": false,
    "pageSize": 100,
    "revalidationInterval": 3600,
    "xjoinHost": "http://localhost:4000/graphql",
  },
  "isMarketplace": false,
  "logging": {
    "cloudwatch": {
      "enabled": false,
      "level": "fatal",
      "options": {
        "aws_access_key_id": "ACCESS_KEY",
        "aws_region": "EU",
        "aws_secret_access_key": "SECRET_ACCESS_KEY",
        "group": "base_app",
        "interval": 1000,
        "prefix": "remediations-",
      },
    },
    "level": "fatal",
    "pretty": true,
  },
  "metrics": {
    "enabled": true,
    "prefix": "remediations_",
    "summaryMaxAge": 600,
  },
  "namespace": "unknown",
  "patchman": {
    "host": "http://localhost:8080",
    "impl": undefined,
    "revalidationInterval": 43200,
  },
  "path": {
    "app": "remediations",
    "base": "/api/remediations",
    "prefix": "/api",
  },
  "platformHostname": "hostname",
  "port": undefined,
  "rbac": {
    "enforce": true,
    "host": "http://localhost:8080",
    "impl": undefined,
    "insecure": false,
  },
  "receptor": {
    "host": "http://localhost:9090",
    "impl": undefined,
  },
  "redis": {
    "enabled": false,
  },
  "requestTimeout": 10000,
  "sources": {
    "host": "http://localhost:8080",
    "impl": undefined,
  },
  "ssg": {
    "host": "http://localhost:8090",
    "impl": undefined,
  },
  "users": {
    "auth": "",
    "clientId": "remediations",
    "env": "prod",
    "host": "https://insights-services-pipeline-insights.ext.paas.redhat.com",
    "impl": undefined,
    "insecure": false,
    "revalidationInterval": 43200,
    "testAccount": "someUsername",
  },
  "validateResponseStrict": true,
  "vmaas": {
    "host": "https://webapp-vmaas-prod.apps.crcp01ue1.o9m8.p1.openshiftapps.com",
    "impl": undefined,
    "revalidationInterval": 43200,
  },
  "vulnerabilities": {
    "auth": "",
    "host": "https://access.qa.itop.redhat.com",
    "impl": undefined,
    "insecure": false,
    "pageSize": 100,
  },
}
`;
